### PR TITLE
Viewing key now is sent as a payload instead registering against enclave

### PR DESCRIPTION
**Pull request by otherview**
### Why this change is needed

https://github.com/obscuronet/obscuro-internal/issues/1610

The enclave no longer stores viewing keys. 
Each request payload is prepended with the viewing key.

### What changes were made as part of this PR

Please provide a high level list of the changes made

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

**Review by zkokelj — APPROVED** [review]
LGTM.

File: integration/simulation/transaction_injector.go
Comment (BedrockSquirrel) [inline integration/simulation/transaction_injector.go:240]: Ha, that's a fun bug.

**Review by BedrockSquirrel — APPROVED** [review]
LGTM, don't love 'handler' names when they're not for handling events but that's just a nitpick.

Please can you run E2E tests if you haven't already, that will test the wallet extension comms thoroughly. Need to make sure the faucet dependencies gets updated to latest obscuro after merge too.

**Pull request merged**
